src/pystack/_pystack/mem.cpp

```diff
@@ -344,42 +344,99 @@ CorefileRemoteMemoryManager::CorefileRemoteMemoryManager(
 {
     CoreFileExtractor extractor{d_analyzer};
     d_shared_libs = extractor.ModuleInformation();
+
+    const char* filename = d_analyzer->d_filename.c_str();
+    int fd = open(filename, O_RDONLY);
+
+    StatusCode ret = readCorefile(fd, filename);
+    close(fd);
+
+    if (ret == StatusCode::ERROR) {
+        throw RemoteMemCopyError();
+    }
+}
+
+CorefileRemoteMemoryManager::~CorefileRemoteMemoryManager()
+{
+    if (munmap(corefile_data, corefile_size) == -1) {
+        LOG(ERROR) << "Failed to un-mmap a file " << d_analyzer->d_filename.c_str();
+    }
+}
+
+CorefileRemoteMemoryManager::StatusCode
+CorefileRemoteMemoryManager::readCorefile(int fd, const char* filename)
```

Comment:
Let's mark this `noexcept` to make it clear that the `close` call before will always be called because this function does not raise